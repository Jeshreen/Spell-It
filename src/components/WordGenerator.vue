<template>
  <div>
    <b-container>
      <b-button @click="getWord">Listen</b-button>
      <b-col>
        <b-row>
          <b-form-input
            v-model="wordCheck"
            type="text"
            id="wordCheck"
          ></b-form-input>
          <b-button id="spellCheck" @click="spellCheck">Spell Check</b-button>
        </b-row>
      </b-col>
      <b-col>
        <b-row>
          <p>{{ spellResults }}</p>
        </b-row>
      </b-col>
    </b-container>
  </div>
</template>

<script>
export default {
  name: "wordGenerator",
  mounted() {
    //ensuring that speech synthesis is supported
    if ("speechSynthesis" in window) {
      console.log("supported");
    } else {
      alert("This browser does not support speech synthesis");
    }

    //trigger the submit functionality on enter
    let wordCheck = document.getElementById("wordCheck");
    wordCheck.addEventListener("keypress", function (event) {
      if (event.key === "Enter") {
        event.preventDefault();
        document.getElementById("spellCheck").click();
      }
    });
  },
  data() {
    return {
      word: "",
      wordCheck: "",
      spellResults: "",
    };
  },
  methods: {
    /**
     * Generating a random word
     */
    getWord() {
      //using the random words npm package to generate a word
      let randomWords = require("random-words");
      this.word = randomWords(1);

      //converting the text to speech
      let speechWord = new SpeechSynthesisUtterance();
      speechWord.rate = 0.5;
      speechWord.text = this.word;
      console.log(this.word);
      window.speechSynthesis.speak(speechWord);
    },

    /**
     * Checking the spelled word
     */
    spellCheck() {
      //validate the word
      let validated = this.validateWord();

      if (validated) {
        return;
      }

      //check if the spelling is correct
      if (this.word != this.wordCheck.toLowerCase()) {
        const errorAudio = new Audio(
          "https://www.fesliyanstudios.com/play-mp3/5265"
        );
        errorAudio.play();
        this.$store.dispatch("incorrectSpelling");
        this.spellResults = "Please try again";
      } else {
        const successAudio = new Audio(
          "https://www.fesliyanstudios.com/play-mp3/5250"
        );
        successAudio.play();
        this.$store.dispatch("successSpelling");
        this.spellResults = "Congratualations!!!";
      }
    },

    /**
     * Method to validate the entered word
     */
    validateWord() {
      if (!this.wordCheck) {
        //Null check
        this.spellResults = "Please Enter the word";
        return true;
      } else if (!this.wordCheck.match(/^[A-Za-z]+$/)) {
        //checking for special characters
        this.spellResults = "Please Enter a valid word with letters only";
        return true;
      } else {
        return false;
      }
    },
  },
};
</script>

<style>
</style>