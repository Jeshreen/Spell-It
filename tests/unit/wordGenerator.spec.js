import { shallowMount } from '@vue/test-utils'
import WordGenerator from '@/components/WordGenerator'

/**
 * Test 1 - Mounting the Number converter component
 */
describe('Mount Component', () => {
  const wrapper = shallowMount(WordGenerator);
  test('does wrapper exists', () => {
    expect(wrapper.exists()).toBe(true)
  })

  it("has the expected html structure", () => {
    expect(wrapper.element).toMatchSnapshot();
  });
})

/**
 * Test 2 - Validation method check
 */
describe('validateWord function test',() => {
  const wrapper = shallowMount(WordGenerator)

  it("null check", async () =>{
    await wrapper.setData({wordCheck: "",spellResults:""})
    wrapper.vm.validateWord();
    expect(wrapper.vm.spellResults).toContain("Please Enter the word")
  })
})