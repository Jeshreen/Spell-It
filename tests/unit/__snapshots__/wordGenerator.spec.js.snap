// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mount Component has the expected html structure 1`] = `
<div>
  <b-container>
    <b-button>
      Listen
    </b-button>
     
    <b-col>
      <b-row>
        <div
          class="letters"
        >
          <b-form-input
            disabled="true"
            id="wordCheck-1"
            maxlength="1"
            type="text"
          />
        </div>
         
        <p>
          
        </p>
         
        <b-button
          disabled="true"
          id="spellCheck"
        >
          Spell Check
        </b-button>
         
        <b-button
          disabled="true"
          id="hint"
        >
          Hint
        </b-button>
      </b-row>
    </b-col>
  </b-container>
</div>
`;
